### ISSUE TITLE
SIGABRT in Aerospike Enterprise Edition build 3.9.0.1 os el6

### ISSUE THREAD

**Issue by RonRothman** [aerospike/aerospike-server#144]
```
Jul 25 2016 22:01:21 GMT: CRITICAL (rw): (rw_request.c:123) rw_request_destroy: origin 3 has non-null 'from'
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:96) SIGABRT received, aborting Aerospike Enterprise Edition build 3.9.0.1 os el6
Jul 25 2016 22:01:21 GMT: INFO (xdr): (xdr.c:5009) XDR is not running.
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: found 15 frames
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 0: /usr/bin/asd(as_sig_handle_abort+0x3c) [0x4c1e6f]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 1: /lib64/libc.so.6(+0x35670) [0x7f91974a1670]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 2: /lib64/libc.so.6(gsignal+0x37) [0x7f91974a15f7]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 3: /lib64/libc.so.6(abort+0x148) [0x7f91974a2ce8]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 4: /usr/bin/asd(generate_column_bits_string+0) [0x59548c]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 5: /usr/bin/asd(rw_request_destroy+0x31) [0x53f311]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 6: /usr/bin/asd(cf_rchash_free+0x26) [0x586355]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 7: /usr/bin/asd(cf_rchash_delete+0xe5) [0x586dcd]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 8: /usr/bin/asd(dup_res_handle_ack+0x3bd) [0x539427]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 9: /usr/bin/asd(rw_msg_cb+0xc7) [0x53ec41]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 10: /usr/bin/asd(fabric_process_read_msg+0x380) [0x510a62]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 11: /usr/bin/asd(fabric_process_readable+0x3c) [0x510ebf]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 12: /usr/bin/asd(fabric_worker_fn+0x2ee) [0x51122a]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 13: /lib64/libpthread.so.0(+0x7dc5) [0x7f9198674dc5]
Jul 25 2016 22:01:21 GMT: WARNING (as): (signal.c:100) stacktrace: frame 14: /lib64/libc.so.6(clone+0x6d) [0x7f9197562c9d]```

**Comment by wchu-citrusleaf** [comment]
Looking at the issue. Thanks.

**Comment by wchu-citrusleaf** [comment]
This "non-null 'from'" assert is recently put in 3.9. Hitting the assert lead us to discover a race condition, where hot key transactions with timeout can prematurely terminate an incoming transaction of the same key. We have fixed this in 3.9.0.2
